layer1/bridge/src/inbox_sc.mligo

```diff
@@ -34,13 +34,17 @@ let make_deposit_message (owner: address) (quantity: nat) : message =
 let make_transaction_message (source: address) (destination: address) (quantity: nat) : message =
   Transaction { source = source ; destination = destination; quantity = quantity; arg = (None : bytes option) }
 
-(** Push a message into the inbox, indexed by the current inbox level **)
-let push_message (max_inbox_level: nat) (inboxes : inboxes) (message: message) : inboxes =
-  let current_level = max_inbox_level in
-  let current_inboxes = inboxes in
-  match Big_map.find_opt current_level current_inboxes with
-  | None -> Big_map.add current_level [ message ] current_inboxes
-  | Some xs -> Big_map.add current_level ( message :: xs ) current_inboxes
+(** When a message is push into an inbox, [current_inbox_level]
+    will be bumped and is used as index of the inbox. One inbox
+    only includes one message.
+
+    [push_message] will return new [inbox_level] and new
+    [inboxes].*)
+let rec push_message (current_inbox_level : nat) (current_inboxes : inboxes) (message: message) : inbox_level * inboxes =
```

Comment:
I'm wondering if a signature like `push_message (message *  state) -> state` would be clearer. What do you think ?

Reply:
I agree. but I'd like keep minimal change here, I tried not to change the signature here.
I create an issue for following up: https://github.com/marigold-dev/chusai/issues/52